doc/whats_new/upcoming_changes/sklearn.tree/32351.fix.rst

```diff
@@ -0,0 +1,3 @@
+- Fix decision tree splitting with missing values present in some features. In some cases the last
+  non-missing sample would not be partitioned correctly.
+  By :user:`Tim Head <betatim>`.
```

Comment:
Let's credit Arthur for the initial analysis, minimal reproducer and concurrent fix in his other PR and contribution about how to write a proper non-regression test for this.

```suggestion
  By :user:`Tim Head <betatim>` and :user:`Arthur Lacote <cakedev0>`.
```